ConsoleApp/Board.cs

```diff
@@ -0,0 +1,177 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace SnakeGame;
+public class Board
+{
+    public static DirectionMap DirectionMap { get; } = new DirectionMap();
+
+    private readonly string instructions = "How to Play: Avoid hitting walls or yourself. Grow by eating food (Y). Highest length wins.";
+    private readonly string commandBase = "Commands: {0}, Esc: Quit\n";
+    private readonly string lengthBase = "Length: {0}\n";
+
+    private Cell[,] grid;
+    private Random random = new Random();
+    private int leftEdgeX => 0;
+    private int rightEdgeX => Width - 1;
+    private int topEdgeY => 0;
+    private int bottomEdgeY => Height - 1;
+
+    public Snake Snake { get; private set; }
+    public int Height { get; private set; }
+    public int Width { get; private set; }
+    public Cell Current => Snake.Head;
+    public Cell Center => get(Width / 2, Height / 2);
+    public Cell Food { get; private set; }
+    public bool HasCollided { get; private set; }
+
+    public Board(int width = 90, int height = 25)
+    {
+        Width = width;
+        Height = height;
+        grid = new Cell[Width, Height];
+        initGrid();
+    }
+
+    public void DoTurn()
+    {
+        doTurn(Snake.Direction, getDestination(Snake.Direction));
+    }
+
+    public void DoTurn(Direction direction)
+    {
+        doTurn(direction, getDestination(direction));
+    }
+
+    public void Draw()
+    {
+        Console.SetCursorPosition(0, 0);
+        Console.WriteLine(ToString());
+    }
+
+    public void Add(Snake snake) => Snake = snake;
+    public void AddFood() => randomCell().SetFood();
+
+    public Cell TopNeighbor(Cell cell) => grid[cell.X, cell.Y - 1];
+    public Cell RightNeighbor(Cell cell) => grid[cell.X + 1, cell.Y];
+    public Cell BottomNeighbor(Cell cell) => grid[cell.X, cell.Y + 1];
+    public Cell LeftNeighbor(Cell cell) => grid[cell.X - 1, cell.Y];
+
+    public override string ToString()
+    {
+        var sb = new StringBuilder();
+        for (int y = 0; y < Height; y++)
+        {
+            for (int x = 0; x < Width; x++)
+            {
+                sb.Append(grid[x, y].Value);
+            }
+            sb.Append("\n");
+        }
+        sb.AppendFormat(lengthBase, Snake.Length);
+        sb.AppendLine(instructions);
+        sb.AppendFormat(commandBase, DirectionMap.ToString());
+        return sb.ToString();
+    }
+
+    private Cell get(int x, int y) => grid[x, y];
```

Comment:
code convention violated
```suggestion
    private Cell Get(int x, int y) => grid[x, y];
```